subprojects/core/src/main/java/org/gradle/api/internal/file/archive/AbstractArchiveFileTree.java

```diff
@@ -36,4 +44,10 @@ public void visitStructure(FileCollectionStructureVisitor visitor, FileTreeInter
             visitor.visitGenericFileTree(owner, this);
         }
     }
+
+    @Override
+    public void visitDependencies(TaskDependencyResolveContext context) {
+        ProviderInternal<File> backingFileProvider = (ProviderInternal<File>) getBackingFileProvider();
+        backingFileProvider.visitDependencies(context);
```

Comment:
```suggestion
        context.add(getBackingFileProvider());
```

That should be enough.